### DISCUSSION TITLE
Dúvida nos arrays

### DISCUSSION THREAD

**Discussion by laurence-martin** [balta-io/2801#65] [Perguntas e Respostas] [answered]
André, qual a diferença inicializar o array das formas abaixo:

var funcionarios = new Funcionarios[5];
//qual a diferença de usar do jeito abaixo
**funcionarios[0] = new Funcionarios(){Id=2795, Name="Andre"}**
//ou abaixo sem usar o new???
**funcionarios[0].Id = 2795;
funcionarios[0].Name = "Andre";**

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @laurence-martin, como vai?

```csharp
var funcionarios = new Funcionarios[5]; // Cria um array

funcionarios[0] = new Funcionarios(){Id=2795, Name="Andre"} // Atribui um item a primeira posição (índice 0) do array

// Só vai funcionar se o item 0 já existir
funcionarios[0].Id = 2795; 
funcionarios[0].Name = "Andre";
```

**Reply by laurence-martin** [reply]
Mas eu testei sem o new Funcionarios e também funcionou, queria entender quando usar o new e quando nao.

**Reply by andrebaltieri** [reply]
Posta seu código completo por favor!

**Comment by laurence-martin** [comment]
var funcionarios = new Funcionarios[5]; // Cria um array
funcionarios[0].Id = 2795; 
funcionarios[0].Name = "Andre";

deste jeito funciona e não precisei usar o new

**Reply by andrebaltieri** [reply]
Boa noite, @laurence-martin, como vai?

Precisou do `new` sim, me referi a primeira linha onde já faz a inicialização do array.
No caso dos arrays dimensionados manualmente, os valores padrão dos tipos sempre são atribuídos.
